src/sentry/stacktraces/processing.py

```diff
@@ -269,42 +267,47 @@ def normalize_stacktraces_for_grouping(data, grouping_config=None):
     # the trimming produces a different function than the function we have
     # otherwise stored in `function` to not make the payload larger
     # unnecessarily.
-    with sentry_sdk.start_span(op=op, description="iterate_frames"):
-        for frames in stacktraces:
+    with sentry_sdk.start_span(op=op, description="iterate_stacktraces"):
+        for frames, exception_data in stacktraces:
             for frame in frames:
-                # Restore the original in_app value before the first grouping
-                # enhancers have been run. This allows to re-apply grouping
-                # enhancers on the original frame data.
-                orig_in_app = get_path(frame, "data", "orig_in_app")
-                if orig_in_app is not None:
-                    frame["in_app"] = None if orig_in_app == -1 else bool(orig_in_app)
-
-                if frame.get("raw_function") is not None:
-                    continue
-                raw_func = frame.get("function")
-                if not raw_func:
-                    continue
-                function_name = trim_function_name(raw_func, frame.get("platform") or platform)
-                if function_name != raw_func:
-                    frame["raw_function"] = raw_func
-                    frame["function"] = function_name
```

Comment:
This whole block is replaced with `_update_frame`:
<img width="750" alt="image" src="https://github.com/getsentry/sentry/assets/44410/66deb216-fb4a-4f0e-9041-dac87ced695c">